From c0c434ea231bbe8e474e8b43896556cdaeff450d Mon Sep 17 00:00:00 2001
From: Xiao,Feng <fengx.xiao@intel.com>
Date: Thu, 10 Jan 2013 11:07:27 +0800
Subject: [PORT FROM MAIN] AVIExtractor: Provide video framerate info

BZ: 46801

For the request form HDMI dynamic setting

Change-Id: I9138bf7e0441bc4a6e8c2900b569fab3825d0ff7
Orig-Change-Id: Id7b5a3a10fa3d53748ccb4d8bdaf66854a80d394
Signed-off-by: Xiao,Feng <fengx.xiao@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/AVIExtractor.cpp |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/AVIExtractor.cpp b/media/libstagefright/AVIExtractor.cpp
index da4fe8e..563e8e6 100644
--- a/media/libstagefright/AVIExtractor.cpp
+++ b/media/libstagefright/AVIExtractor.cpp
@@ -935,6 +935,13 @@ status_t AVIExtractor::parseIndex(off64_t offset, size_t size) {
         AString mime = tmp;
 
         if (!strncasecmp("video/", mime.c_str(), 6)) {
+            if (durationUs != 0) {
+                int32_t frameRate = (track->mSamples.size() * 1000000LL + (durationUs >> 1)) / durationUs;
+                if (frameRate > 0) {
+                    track->mMeta->setInt32(kKeyFrameRate, frameRate);
+                    ALOGV("video framerate = %d", frameRate);
+                }
+            }
             if (track->mThumbnailSampleIndex >= 0) {
                 int64_t thumbnailTimeUs;
                 CHECK_EQ((status_t)OK,
-- 
1.7.4.1

